src/pk.c

```diff
@@ -51,6 +51,487 @@ static int pem_write_bio_pubkey(WOLFSSL_BIO* bio, WOLFSSL_EVP_PKEY* key);
 
 #if defined(OPENSSL_EXTRA)
 
+#if (!defined(NO_DH) && (defined(HAVE_LIGHTY) || defined(HAVE_STUNNEL) || \
+     defined(WOLFSSL_MYSQL_COMPATIBLE) || defined(OPENSSL_EXTRA))) || \
+    (defined(OPENSSL_EXTRA) && !defined(WOLFCRYPT_ONLY))
+#ifndef NO_BIO
+
+#ifdef WOLFSSL_NO_FSEEK
+/* Amount of memory to allocate/add. */
+#define READ_BIO_FILE_CHUNK     128
+
+/* Read a file in chunks.
+ *
+ * Allocates a chunk and reads into it until it is full.
+ *
+ * @param [in, out] bio  BIO object to read with.
+ * @parma [out]     pem  Read data in a new buffer.
+ * @return  -ve on error.
+ * @return  Number of bytes read on success.
+ */
+static int wolfssl_read_bio_file(WOLFSSL_BIO* bio, char** pem)
+{
+    int ret = 0;
+    char* mem;
+    char* p;
+
+    /* Allocate buffer to hold a chunk of data. */
+    mem = (char*)XMALLOC(READ_BIO_FILE_CHUNK, bio->heap, DYNAMIC_TYPE_OPENSSL);
+    if (mem == NULL) {
+        WOLFSSL_MSG("Memory error");
```

Comment:
Should we use WOLFSSL_ERROR_MSG? It's not often used, but might be a good time to start.

Reply:
Yep. Good to have error log levels finally!
Fixed RSA and DH APIs.